### PR TITLE
Improve file associations

### PR DESCRIPTION
**Pull request by United600**
Add file extension logos and localized names.

App default apps settings

I couldn't reliably add the `.ttf` extension, sometimes it would refuse to compile. I was pretty sure those [restrictions](https://learn.microsoft.com/en-us/windows/uwp/launch-resume/reserved-uri-scheme-names#file-types-reserved-for-the-operating-system) were relaxed but maybe I have the wrong SDK version.

**Samples**

https://github.com/user-attachments/assets/d9930080-2a68-4a83-8fb7-9681ae0d0281

https://github.com/user-attachments/assets/b25434c5-d1fa-47c0-89dc-14365a3d8efb

https://github.com/user-attachments/assets/833b4a7e-1662-4b7a-b6f1-c193f4e947a5

### CODE REVIEW

**Comment by JohnnyWestlake** [comment]
Unfortunately will probably have to drop this one unless we  provide a way to install fonts - the  problem is the default Windows Font Viewer is used as the primary way to install fonts on the system and once we remove the default association we remove the default way for people to install fonts - to the point that various software vendors and font distributors used to have support articles recommending users uninstall Character Map UWP entirely so they could get their fonts working. There is a large non-technical user base of the app who use it for design and craft work and getting them workaround to install fonts doesn't really work.

So, basically, unless we provide a font install method ourselves, we can't really do this :') And to do _that_, we need to be able to write to the registry. Maybe there's a way to do it now, I haven't tried in a few years.

**Comment by United600** [comment]
> Unfortunately will probably have to drop this one unless we provide a way to install fonts

Unfortunate indeed. Sorry, didn't knew.
Yeah if I set it as the default app, the `Install` verb disappears. That's not good.
  
ContextMenu

I don't think setting [desktop2:AllowSilentDefaultTakeOver](https://learn.microsoft.com/en-us/uwp/schemas/appxpackage/uapmanifestschema/element-uap-filetypeassociation) will make a difference since, even without it, it only takes over the `.woff`/`.woff2` extensions (oh and the `.otc`).

> So, basically, unless we provide a font install method ourselves, we can't really do this :') And to do _that_, we need to be able to write to the _real_ registry. Maybe there's a way to do it now, I haven't tried in a few years.

AFAIK it is possible, but needs the [unvirtualized](https://learn.microsoft.com/en-us/uwp/schemas/appxpackage/uapmanifestschema/element-desktop6-registrywritevirtualization) registry [rescap](https://learn.microsoft.com/en-us/windows/uwp/packaging/app-capability-declarations). Personally, I don't like that tradeoff. More [info](https://learn.microsoft.com/en-us/windows/msix/desktop/flexible-virtualization).
 
> (Fwiw, `.ttf` is still in the operating-system reserved file type list - https://learn.microsoft.com/en-us/windows/uwp/launch-resume/reserved-uri-scheme-names#file-types-reserved-for-the-operating-system )

Yes but, IIRC as of Windows 10 2004, those restrictions were "lifted"/"relaxed".

I can limit the associations to the current ones for now.

**Comment by United600** [comment]
> the problem is the default Windows Font Viewer is used as the primary way to install fonts on the system and once we remove the default association we remove the default way for people to install fonts - to the point that various software vendors and font distributors used to have support articles recommending users uninstall Character Map UWP entirely so they could get their font installs working (which is why I had to remove them previously). There is a large non-technical user base of the app who use it for design and craft work and getting them workaround to install fonts doesn't really work.

Although I can see the problem, the ability to "register" the app on the `Open with` menu is really convenient but, unknowingly, the user can set it as the default app and that will cause the issues that you highlight there. Even though its easily reversible.

Can we add a Button/MenuFlyoutItem to launch the [menu](https://learn.microsoft.com/en-us/windows/uwp/launch-resume/launch-the-default-app-for-a-file#open-with-launch) so they can reverse or open the Font Viewer to install?

**Comment by JohnnyWestlake** [comment]
> Can we add a Button/MenuFlyoutItem to launch the [menu](https://learn.microsoft.com/en-us/windows/uwp/launch-resume/launch-the-default-app-for-a-file#open-with-launch) so they can reverse or open the Font Viewer to install?

The problem is I mostly work on this for people in the design / crafting community who are not very well tech versed and asking them to change their default file association ends up being harder than them uninstalling it, and the support companies for these devices they're using would rather recommend they uninstall it as it's not their problem to deal with.

I'm not personally opposed to the unvirtualized registry rescap, I'll take another look at it. It will only allow install to the current user profile, but it might be good enough.

**Comment by JohnnyWestlake** [comment]
On this front, per-user fonts should be registered at `HKEY_CURRENT_USER\Software\Microsoft\Windows NT\CurrentVersion\Fonts`, which we could access with the above.

But we would need to copy the files to a file path that would survive the application getting uninstalled. Ideally they would get registered at `%USERPROFILE\AppData\Local\Microsoft\Windows\Fonts`, but we can't write there by default. And if we write to our own local storage (or publisher cache), they'll all get deleted on uninstall

**Comment by United600** [comment]
AFAIK, per-user install as been the default for some time.

> But we would need to copy the files to a file path that would survive the application getting uninstalled. Ideally they would get registered at %USERPROFILE%\AppData\Local\Microsoft\Windows\Fonts, but we can't write there by default. And if we write to our own local storage (or publisher cache), they'll all get deleted on uninstall. So need to find somewhere persistent that we can access.

Just checked on how the Store handles that and it's not an option. Persist uninstall will be hard and would have to place them on a "hidden/hard to find" place to prevent accidental deletion.

Registry

Microsoft 365 doesn't even register the cloud fonts with the system.

File explorer font cache
